CONTRIBUTING.adoc

```diff
@@ -1,232 +1,140 @@
+
+Many issues are caused by subtle behavior, typos, and unintended configuration.
+Creating a https://stackoverflow.com/help/minimal-reproducible-example[Minimal Reproducible Example] (starting with https://start.spring.io for example) of the problem helps the team quickly triage your issue and get to the core of the problem.
+
+We love contributors, and we may ask you to <<submit-a-pull-request,submit a PR with a fix>>.
+
+[[issue-lifecycle]]
+=== Issue Lifecycle
+
+When an issue is first created, it is flagged `waiting-for-triage` waiting for a team member to triage it.
+Once the issue has been reviewed, the team may ask for further information if needed, and based on the findings, the issue is either assigned a target branch (or no branch if a feature) or is closed with a specific status.
+The target branch is the earliest supported branch where the change will be applied.
+
+When a fix is ready, the issue is closed and may still be re-opened until the fix is released.
+After that the issue will typically no longer be reopened.
+In rare cases if the issue was not at all fixed, the issue may be re-opened.
+In most cases however any follow-up reports will need to be created as new issues with a fresh description.
+
+[[submit-a-pull-request]]
+=== Submit a Pull Request
+
+We are excited for your pull request! :heart:
+
+Please do your best to follow these steps.
+Don't worry if you don't get them all correct the first time, we will help you.
+
+[[sign-cla]]
+1. If you have not previously done so, please sign the
+https://cla.spring.io/sign/spring[Contributor License Agreement].
+You will be reminded automatically when you submit the PR.
+[[create-an-issue]]
+1. Must you https://github.com/spring-projects/spring-security/issues/new/choose[create an issue] first? No, but it is recommended for features and larger bug fixes. It's easier discuss with the team first to determine the right fix or enhancement.
+For typos and straightforward bug fixes, starting with a pull request is encouraged.
+Please include a description for context and motivation.
+Note that the team may close your pull request if it's not a fit for the project.
+[[choose-a-branch]]
+1. Always check out the branch indicated in the milestone and submit pull requests against it.
+If there is no milestone, choose `main`.
+Once merged, the fix will be forwarded-ported to applicable branches including `main`.
+[[create-a-local-branch]]
+1. Create a local branch.
+If this is for an issue, consider a branch name with the issue number, like `gh-22276`.
+[[write-tests]]
+1. Add JUnit Tests for your changes
+[[add-since]]
+1. If on `main`, add `@since` JavaDoc attributes to new public APIs that your PR adds
+[[change-rnc]]
+1. If you are updating the XSD, please instead update the RNC file and then run `./gradlew :spring-security-config:rncToXsd`.
+[[format-code]]
+1. For each commit, build the code using `./gradlew check format`.
+This command ensures the code meets the style guide.
```

Comment:
Unfortunately, the `format` task does not organize imports, maybe linking to the "Source Code Style" section in case of import order errors could help